addons/website_quote/models/order.py

```diff
@@ -1,229 +1,215 @@
-    }
-    def on_change_product_id(self, cr, uid, ids, product, context=None):
-        vals = {}
-        product_obj = self.pool.get('product.product').browse(cr, uid, product, context=context)
-        name = product_obj.name
-        if product_obj.description_sale:
-            name += '\n' + product_obj.description_sale
-        vals.update({
-            'price_unit': product_obj.lst_price,
-            'product_uom_id': product_obj.uom_id.id,
-            'website_description': product_obj and (product_obj.quote_description or product_obj.website_description) or '',
-            'name': name,
-        })
-        return {'value': vals}
-
-    def _inject_quote_description(self, cr, uid, values, context=None):
+
+    sequence = fields.Integer(help="Gives the sequence order when displaying a list of sale quote lines.", default=10)
+    quote_id = fields.Many2one('sale.quote.template', string='Quotation Template Reference', required=True, ondelete='cascade', index=True)
+    name = fields.Text(string='Description', required=True, translate=True)
+    product_id = fields.Many2one('product.product', string='Product', domain=[('sale_ok', '=', True)], required=True)
+    website_description = fields.Html(string='Line Description', related="product_id.product_tmpl_id.quote_description", translate=True)
+    price_unit = fields.Float(string='Unit Price', required=True, digits_compute=dp.get_precision('Product Price'))
+    discount = fields.Float(digits_compute=dp.get_precision('Discount'))
+    product_uom_qty = fields.Float(string='Quantity', required=True, digits_compute=dp.get_precision('Product UoS'), default=1)
+    product_uom_id = fields.Many2one('product.uom', string='Unit of Measure ', required=True)
+
+    @api.onchange('product_id')
+    def on_change_product_id(self):
+        product = self.product_id
+        name = product.name
+        self.price_unit = product.list_price
+        self.product_uom_id = product.uom_id.id
+        self.website_description = product and (product.quote_description or product.website_description) or ''
+        self.name = product.description_sale and name + '\n' + product.description_sale or name
+
+    def _inject_quote_description(self, values):
         values = dict(values or {})
         if not values.get('website_description') and values.get('product_id'):
-            product = self.pool['product.product'].browse(cr, uid, values['product_id'], context=context)
+            product = self.env['product.product'].browse(values['product_id'])
             values['website_description'] = product.quote_description or product.website_description or ''
         return values
 
-    def create(self, cr, uid, values, context=None):
-        values = self._inject_quote_description(cr, uid, values, context)
-        ret = super(sale_quote_line, self).create(cr, uid, values, context=context)
+    @api.model
+    def create(self, values):
+        values = self._inject_quote_description(values)
```

Comment:
yup.Remove _inject_quote_description() and make website_description a computed field.